### PR TITLE
Factor out common stack adjustment code

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test this please

**Comment by BruceForstall** [comment]
@mmitche I don't see the testing kicked off for this PR. Is the CI processing PR testing now? Am I just being impatient?

File: src/jit/emitxarch.cpp [hunk 1]
```diff
@@ -5196,19 +5042,76 @@ void emitter::emitIns_J(instruction ins, BasicBlock* dst, int instrCount /* = 0
...
+}
 
-#endif // !FEATURE_FIXED_OUT_ARGS
+//------------------------------------------------------------------------
+// emitAdjustStackDepth: Adjust the current and maximum stack depth.
+//
+// Arguments:
+//    ins - the instruction. Only INS_add and INS_sub adjust the stack depth.
+//          It is assumed that the add/sub is on the stack pointer.
+//
```
Comment (bendono) [inline src/jit/emitxarch.cpp]: Missing a description of the val argument.

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib PTAL

**Comment by pgavlin** [comment]
LGTM

**Pull request merged**